components/script/dom/vrpose.rs

```diff
@@ -33,9 +33,13 @@ unsafe fn update_or_create_typed_array(cx: *mut JSContext,
     match src {
         Some(ref data) => {
             if dst.get().is_null() {
-                dst.set(slice_to_array_buffer_view(cx, &data));
+                let _ = TypedArray::<Float32>::create(cx, data.len() as u32, src,
+                    dst.handle_mut());
```

Comment:
nit: `Float32Array::create`, and move `dst.handle_mut()` into the previous line.